### ISSUE TITLE
navigation buttons optional property - buttons missing from the options

### ISSUE THREAD

**Issue by ronm123** [thexerteproject/xerteonlinetoolkits#574]
With the default navigation set to anything but linear we have some buttons missing from the options when changing the navigation button options for a particular page. e.g. at the moment we have the following:
All
Back and Next
Back only
Next only
None

However with navigation set to historic there isn't an option to include or exclude the home button
and with navigation set to menu or menu with page controls (or indeed linear) there's no option to include or exclude the contents button which in the case of the 2 menu options is what goes back to the menu on the first page. But I guess adding these to the navigation button options isn't straightforward because control over each additional button add at least 3 extra permutations. e.g. just to add the home button in the mix (which would be useful) I think we would need:
All
Home only
Home, Back and Next
Back and Next only
Home and Back only
Back only
Home and Next only
Next only
None
And then even more if you add Contents to the mix.

I know we can show/hide via jquery and/or css but just wondering if there's a way to enable this for everyone but also in a way that doesn't require so many permutations in the list menu?

**Comment by FayCross** [comment]
How about we change it so that it uses the new group of optional properties instead of just 1 drop down. You also wouldn't need separate options for home & contents etc. as I think when the home button's there the content button is normally not. The options would look something like this...

Navigation buttons:
- [ ] Home / Contents / Menu
- [ ] Back / Previous
- [ ] Next

**Comment by ronm123** [comment]
@FayCross sounds like a good idea. Assuming that change wouldn't break existing LO's that have used the dropdown options?

**Comment by FayCross** [comment]
Ok, I'll look into it

**Comment by FayCross** [comment]
@ronm123 Just realised I assigned this to myself months ago - what I've just committed shouldn't break existing projects but give a bit more flexibility as to which buttons can be turned on / off

**Comment by ronm123** [comment]
@FayCross great I'll give it a try.
By coincidence I was looking at some projects yesterday and realised and recalled that it would also be quite useful to add an optional property to include a home button when using linear navigation and perhaps with control over what page that actually goes to. I know the TOC button allows that but obviously that's two clicks and less intuitive. What do you think is it worth adding that as an enhancement issue for future development?

**Comment by FayCross** [comment]
Yes we could add something like that too. I was wondering also about the option to show / hide the page numbers when it's linear. At the moment when adding a progress bar you can choose to hide the numbers but it would give greater flexibility to have progress bar & page numbers as two separate options.

Much more complicated but I was also thinking about how it would be useful when navigation is linear to allow for pages to be taken out of the main flow (& page numbering) and only accessed via links from other pages. So, for example, you could have a link from a page to something like a 'more info' page which wouldn't be numbered & wouldn't have a next button but the back button would work as a historical back to take you back to the main strand. I guess it would be similar to hide page option but you could actually access the hidden pages from links if needed.

**Comment by ronm123** [comment]
@FayCross I'm probably forgetting how this worked previously and so thes issue may have existed already but looking at the table of contents group of properties it seems like there's a bug with some of this.

If I set the LO navigation to menu or menu with page controls and set the page image option to 'Use Panel' I get an image in a panel on the menu page and using the buttons to navigate to a page shows the content of that destination page. So I think that's how it should be except that the image is displayed below the buttons and I seem to recall it used to display to the right of the buttons?

If I set the page image to 'Full Screen' rather than 'Use Panel' I don't get an image on the menu page on first playing the LO but the image becomes a full screen background to all the pages. If I then click the TOC button to go back to the menu page the 'Full Screen' image is then shown on the menu page but also remains on all the pages. That image also overrides any image that was set using the separate LO background image option e.g. the background image doesn't show with the TOC image set to full screen. Clearly the TOC image should only show on the menu page.

Also if I have lots of pages e.g. 16 so that the buttons need to scroll on first play that scrolling is hidden but then shown after navigating to a page and then back to the menu.
